### ISSUE TITLE
Roundtrip bug in BLS public key aggregation

### ISSUE THREAD

**Issue by tobowers** [dedis/kyber#400]
When aggregating public keys and round tripping them to binary, there's an odd bug where it fails if the first key is repeated. It succeeds in every other order I can find. You can see a failing test here: https://github.com/dedis/kyber/compare/v3.0.8...quorumcontrol:bug/aggregate

If you have public1,public2,public3

If you aggregate: public1,public1,public2 it will fail, but public1,public2,public1 succeeds as well as public1,public2,public2 - so it seems that it only fails when repeated first in the sequence.

**Comment by jeffallen** [comment]
The point is already not on the curve (i.e. invalid) after the second consecutive point Add:
```
func Test_g2_2add_oncurve(t *testing.T) {
	s := NewSuite()
	p := s.G2().Point()
	x := s.G2().Point().Pick(random.New())
	//y := s.G2().Point().Pick(random.New())
	p.Add(p, x)
	require.True(t, p.(*pointG2).g.IsOnCurve())
	p.Add(p, x)
	require.True(t, p.(*pointG2).g.IsOnCurve())
}
```

I showed this to @Daeinar who brought this code to us, and he said "wow". We are now investigating if cloudflare/bn256, which we derive from, has the same problem (or a fix).

**Comment by jeffallen** [comment]
Just an update on this: I have been applying the patches from Cloudflare's solution to this problem, and though I see how it solves the problem in their library, it doesn't solve the problem in ours, and in fact makes additional tests of ours fail. Still digging.

**Comment by jeffallen** [comment]
This is released in v3.0.9.
